### PR TITLE
Add an error return to r_core_project_cat ##projects

### PR DESCRIPTION
**Pull request by RHL120**
- [X] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/core/project.c [hunk 1]
```diff
@@ -87,16 +87,18 @@ R_API bool r_core_is_project(RCore *core, const char *name) {
 }
 
 R_API int r_core_project_cat(RCore *core, const char *name) {
```
Comment (trufae) [inline libr/core/project.c:89]: I think it should return bool. The command handler who calls this function is the one that must change the return code

**Comment by trufae** [comment]
Please rebase and try again when https://github.com/radareorg/radare2/pull/19897 gets merged

**Comment by trufae** [comment]
I would like to move core/rvc.c into util/vc.c and rename include/rvc.h to include/r_util/vc.h

the only blocker for this seems to be only the dependency on r_hash. So i propose making this a function pointer that RCore can replace to substitute the default algorithm. Ideally there should be always the same checksum algorithm used everywhere.. so i'm not fully convinced about this solution. what do you think?

File: libr/core/project.c [hunk 2]
```diff
@@ -86,17 +86,18 @@ R_API bool r_core_is_project(RCore *core, const char *name) {
...
-R_API int r_core_project_cat(RCore *core, const char *name) {
+R_API void r_core_project_cat(RCore *core, const char *name) {
+	core->num->value = 1;
 	char *path = get_project_script_path (core, name);
 	if (path) {
 		char *data = r_file_slurp (path, NULL);
 		if (data) {
 			r_cons_println (data);
 			free (data);
+			core->num->value = 0;
```
Comment (trufae) [inline libr/core/project.c:97]:
```suggestion
			r_core_return_code (core, R_CMD_RC_SUCCESS);
```

File: libr/core/project.c [hunk 3]
```diff
@@ -86,17 +86,18 @@ R_API bool r_core_is_project(RCore *core, const char *name) {
 	return ret;
 }
 
-R_API int r_core_project_cat(RCore *core, const char *name) {
+R_API void r_core_project_cat(RCore *core, const char *name) {
+	core->num->value = 1;
```
Comment (trufae) [inline libr/core/project.c:90]:
```suggestion
				r_core_return_code (core, R_CMD_RC_FAILURE);
```

**Review by trufae — CHANGES_REQUESTED** [review]
pls merge those changes and rebase

**Comment by Lazula** [comment]
The test `Save PE project with maddr` will be fixed by #19965

**Pull request merged**